src/App.tsx

```diff
@@ -1,204 +1,200 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import * as todoService from './api/todos';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
+
+function getFilteredTodos(
+  currentTodos: Todo[],
+  setCurrentFilter: 'all' | 'active' | 'completed',
+) {
+  const filteredTodos = [...currentTodos];
+
+  switch (setCurrentFilter) {
+    case 'active':
+      return filteredTodos.filter(todo => !todo.completed);
+    case 'completed':
+      return filteredTodos.filter(todo => todo.completed);
+    case 'all':
+      return filteredTodos;
+    default:
+      return;
```

Comment:
```suggestion
    case 'all':
    default:
      return filteredTodos;
```
always return an array

Or `throw new Error('Invalid filter')` - it should never happen, just for consistency and type safety, it's a common approach